src/mgr/DaemonServer.cc

```diff
@@ -548,24 +548,16 @@ bool DaemonServer::handle_close(const ref_t<MMgrClose>& m)
   return true;
 }
 
-void DaemonServer::update_task_status(DaemonKey key, const ref_t<MMgrReport>& m) {
-  dout(10) << "got task status from " << key << dendl;
-
-  bool service_map_dirty = false;
-  if ((*m->task_status).empty()) {
-    auto removed = pending_service_map.rm_daemon(key.type, key.name);
-    if (removed) {
-      service_map_dirty = true;
-    }
-  } else {
-    auto p = pending_service_map.get_daemon(key.type, key.name);
-    if (!map_compare(p.first->task_status, *m->task_status)) {
-      service_map_dirty = true;
-      p.first->task_status = *m->task_status;
-    }
-  }
+void DaemonServer::update_task_status(
+  DaemonKey key,
+  const std::map<std::string,std::string>& task_status)
+{
+  derr << "got task status from " << key << dendl;
 
-  if (service_map_dirty) {
+  [[maybe_unused]] auto [daemon, added] =
+    pending_service_map.get_daemon(key.type, key.name);
+  if (daemon->task_status != task_status) {
+    daemon->task_status = task_status;
```

Comment:
Looks like this would result in empty `task status` in ceph status`? (i.e., no entries in task status field). We probably want to only dump if there is something to be shown?

Reply:
i think you are missing the point of this change. service_map is not for tracking "task_status". its consumer uses it to understand if the cluster has a certain service or not.

Reply:
and the "empty `task status` in `ceph status` is not in the scope of this change. we could address it in another PR though, if that concerns you.

Reply:
I do understand the point. However, task_status is a part of service_map. Removing a service from service map was resulting in this issue -- I kind of assumed that normal ceph services would not be a part of service map.

Also, the point of empty task_status -- I'm ok with fixing that as a separate pr.

Reply:
#36139